api/src/main/java/org/openmrs/module/pihcore/task/PihRemovePatientsFromMCOEQueue.java

```diff
@@ -88,11 +88,14 @@ private int removeTimedOutPatients(Queue queue) {
         for (QueueEntry entry : queueEntries) {
             Date startedAt = entry.getStartedAt();
             long activeMinutes = Duration.between(startedAt.toInstant(), currentTime.toInstant()).toMinutes();
-            log.warn("Removing patient {} from queue after {} minutes", entry.getPatient().getUuid(), activeMinutes);
             entry.setEndedAt(currentTime);
-            queueServices.getQueueEntryService().saveQueueEntry(entry);
-            numPatientsRemoved++;
-
+            try {
+                queueServices.getQueueEntryService().saveQueueEntry(entry);
+                log.warn("Removing patient {} from queue after {} minutes", entry.getPatient().getUuid(), activeMinutes);
```

Comment:
Not sure why you moved this log statement here?

Reply:
I mean from the original code @cioan .  the log statement says "removing", not "removed", which implies the logging is coming prior to the operation.  I am not sure why you moved this to happen _after_ the save.  We can move it there I guess, but then I would expect the message to say "Removed patient ...".